apps/client/src/app/codesystem/codesystem.component.ts

```diff
@@ -158,7 +158,7 @@ export class CodesystemComponent extends BaseComponent implements OnInit, OnDest
       return;
     }
 
-    this.codeSystemService.save(this.codeSystem)
+    this.codeSystemService.save(this.codeSystem, this.getCodeSystemID())
```

Comment:
I'm thinking the changes to `codesystem.component.ts` and `codesystem.service.ts` are not necessary... Can you confirm?

Reply:
The issue here is that the codesystem resource doesn't store the ID within it's data object.  When it goes through save and does its check for the ID, it was completely erroring out and properly saving the codesystem. This fix corrects that.

Reply:
Removed